### DISCUSSION TITLE
Don't require verbatim string modifier '@' on multi-line string when chop lines are all in the interpolated part

### DISCUSSION THREAD

**Discussion by kyoyama-kazusa** [dotnet/csharplang#4276] [Language Ideas]
Before:
```csharp
public override string ToString() =>
    $@"r{Cell / 9 + 1}c{Cell % 9 + 1} {ConclusionType switch
    {
        ConclusionType.Assignment => "=",
        ConclusionType.Elimination => "<>"
    }} {Digit + 1}";
```

After:
```csharp
public override string ToString() =>
    $"r{Cell / 9 + 1}c{Cell % 9 + 1} {ConclusionType switch
    {
        ConclusionType.Assignment => "=",
        ConclusionType.Elimination => "<>"
    }} {Digit + 1}";
```

The interpolated part `ConclusionType switch { ConclusionType.Assignment => "=", ConclusionType.Elimination => "<>" }` chops   lines, but other character won't chop lines.

**Comment by YairHalberstadt** [comment]
Why do you not want to use `@`

**Reply by kyoyama-kazusa** [reply]
`@` mark can maintain the characters original, in this way we can reduce the misuse on escape characters; if we found that all characters are original, `@` will be unnecessary. However, the above example shows that the string is **unnecessary** to be marked as a verbatim string because all characters in the string are normal. If I remove `@`, compiler will tell me this is an error. Therefore, I think this is a syntax bug.

**Reply by theunrepentantgeek** [reply]
It's not a syntax bug - for a regular (no `@`) string, the entire string must be on the same line _before_ the semantics of `$` are considered.

FWIW, If I saw your sample code in a code review, I would block merging until you introduced a couple of intention revealing well named local variables:

``` csharp
public override string ToString() 
{
    var mystery = Cell % 9 + 1;
    var conclude = ConclusionType switch
    {
        ConclusionType.Assignment => "=",
        ConclusionType.Elimination => "<>"
    };
    return $"r{mystery}c{mystery} {conclude} {Digit + 1}";
}
```
Both `mystery` and `conclude` need better names - names that actually tell the person reading the code what they mean.

Shmooshing every possible thing into a single expression doesn't make code easier to read nor to maintain.

**Reply by kyoyama-kazusa** [reply]
@theunrepentantgeek Oh! I'm wrong! 😞 
In fact, the example is extracted from my sudoku game repo. A cell means a sudoku cell to fill the digit 1 to 9, and the conclusion type means either the cell should be filled with that digit, or remove the possibility of that digit.
Anyway, thanks for your reply!
